utils-zookeeper/src/main/java/cz/o2/proxima/utils/zookeeper/ZKGlobalWatermarkTracker.java

```diff
@@ -152,15 +150,22 @@ private int getSessionTimeout(Map<String, Object> cfg) {
         .orElse(60000);
   }
 
-  private URI getZkUri(Map<String, Object> cfg) {
-    URI uri =
+  @VisibleForTesting
+  void parseZkUri(Map<String, Object> cfg) {
```

Comment:
I don't think we have to change the method contract here.

Can we just use `uri.getAuthority()` (authority represents host:port combinations) instead "manual" solutions?
```
    String.format("%s:%d", uri.getHost(), uri.getPort());
```
```
    zkConnectString = stringUri.substring(zkPrefix.length(), stringUri.indexOf(uri.getPath()));
```

Reply:
Good point, it leads to significantly smaller code modification.